### PR TITLE
Documentation of programmatic configuration to XML translation feature

### PR DESCRIPTION
**Pull request by albinsuresh**

### CODE REVIEW

File: docs/src/docs/asciidoc/user/xml.adoc [hunk 1]
```diff
@@ -129,3 +129,35 @@ include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[ta
...
+include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[tag=xmlTranslation]
+----
+<1> Instantiate an `XmlConfiguration` passing the cache manager `Configuration`
+<2> Retrieve the XML representation using the `toString` method.
+
+Not every programmatic configuration can be translated to their XML equivalent in this manner.
+Translation is not supported if cache manager configuration contains a cache with any one of the following configured:
+
+. `EvictionAdvisor`
+. A custom `ExpiryPolicy` other than `timeToLiveExpiration` and `timeToIdleExpiration` from ExpiryPolicyBuilder`
```
Comment (ChrisBradley001) [inline docs/src/docs/asciidoc/user/xml.adoc]: and -> or

File: docs/src/docs/asciidoc/user/xml.adoc [hunk 2]
```diff
@@ -129,3 +129,35 @@ include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[ta
...
+
+[source,java,indent=0]
+----
+include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[tag=xmlTranslation]
+----
+<1> Instantiate an `XmlConfiguration` passing the cache manager `Configuration`
+<2> Retrieve the XML representation using the `toString` method.
+
+Not every programmatic configuration can be translated to their XML equivalent in this manner.
+Translation is not supported if cache manager configuration contains a cache with any one of the following configured:
```
Comment (ChrisBradley001) [inline docs/src/docs/asciidoc/user/xml.adoc]: if cache manager -> if the cache manager

File: docs/src/docs/asciidoc/user/xml.adoc [hunk 3]
```diff
@@ -129,3 +129,35 @@ include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[ta
...
+<2> The inherent properties can be overridden by simply providing a different value prior to building the `CacheConfiguration`
+
+[source,java,indent=0]
+----
+include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[tag=xmlTranslation]
+----
+<1> Instantiate an `XmlConfiguration` passing the cache manager `Configuration`
+<2> Retrieve the XML representation using the `toString` method.
+
+Not every programmatic configuration can be translated to their XML equivalent in this manner.
```
Comment (ChrisBradley001) [inline docs/src/docs/asciidoc/user/xml.adoc]: their -> its

File: docs/src/docs/asciidoc/user/xml.adoc [hunk 4]
```diff
@@ -129,3 +129,35 @@ include::{sourcedir35}/xml/src/test/java/org/ehcache/docs/GettingStarted.java[ta
 ----
 <1> Creates a builder, inheriting the capacity constraint of 200 entries
 <2> The inherent properties can be overridden by simply providing a different value prior to building the `CacheConfiguration`
+
+== Programmatic configuration to XML
+
+Just like getting a cache manager configuration from its xml equivalent, the reverse translation is supported too.
```
Comment (ChrisBradley001) [inline docs/src/docs/asciidoc/user/xml.adoc]: xml -> XML

**Comment by henri-tremblay** [comment]
@albinsuresh Isn't it weird that this is at the "getting started" level?

**Comment by albinsuresh** [comment]
@henri-tremblay I'm confused. It *IS* currently in the XML specific `xml.adoc` file. And the sample code is also in the xml specific `GettingStarted` class and *NOT* in the main `GettingStarted` class. You wanted it elsewhere?

**Comment by henri-tremblay** [comment]
No. My bad. I read too quickly...

**Comment by albinsuresh** [comment]
test this please

**Pull request merged**